compiler bug: compiler confused by function called vbytes() calling array.vbytes()

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 12.0.1, 21A559
Processor: 10 cpus, 64bit, little endian, Apple M1 Max
CC version: Apple clang version 13.0.0 (clang-1300.0.29.3)

getwd: /Users/thomas/bug
vmodules: /Users/thomas/.vmodules
vroot: /Users/thomas/Unix/v-latest
vexe: /Users/thomas/Unix/v-latest/v
vexe mtime: 2021-11-29 09:04:46
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 24ba660.c14c324

Git version: git version 2.34.0
Git vroot status: weekly.2021.47-49-gc14c3241
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg bug.v`
```v
type Buffer = []byte

pub fn (mut sb Buffer) vbytes() string {
	return sb[0..sb.len].vbytes().str()
}

fn main() {
	mut b := Buffer([]byte{cap:10})
	b << 1
	b << 2
	println(b.vbytes())
}

```

**What did you expect to see?**

no crash


**What did you see instead?**
```
==================
       ^
/tmp/v_501/bug.10506693758119739925.tmp.c:10602:26: error: passing 'int' to parameter of incompatible type 'string' (aka 'struct string')
        string _t1 = string_str(main__Buffer_vbytes_static(array_slice(*sb, 0, sb->len)));
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/v_501/bug.10506693758119739925.tmp.c:9362:26: note: passing argument to parameter 's' here
string string_str(string s) {
                         ^
2 errors generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

`string _t1 = string_str(byteptr_vbytes(&array_slice(*sb, 0, sb->len)));`

```
cc: /tmp/v_1000/bug.01J8GVC9EWGNTR2VFHP39SKJHC.tmp.c:13334: warning: assignment from incompatible pointer type
cc: /tmp/v_1000/bug.01J8GVC9EWGNTR2VFHP39SKJHC.tmp.c:13334: error: too few arguments to function
```

We must fix this case... since byteptr_vbytes has 2 params.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1675,13 +1675,15 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 	// if so, then instead of calling array_clone(&array_slice(...))
 	// call array_clone_static(array_slice(...))
 	mut is_range_slice := false
-	if node.name == 'clone' && node.receiver_type.is_ptr() && !left_type.is_ptr() {
+	if node.receiver_type.is_ptr() && !left_type.is_ptr() {
 		if node.left is ast.IndexExpr {
 			idx := node.left.index
 			if idx is ast.RangeExpr {
 				// expr is arr[range].clone()
 				// use array_clone_static instead of array_clone
-				name = util.no_dots('${receiver_type_name}_${node.name}_static')
+				if node.name == 'clone' {
+					name = util.no_dots('${receiver_type_name}_${node.name}_static')
+				}
 				is_range_slice = true
 			}
 		}
@@ -1758,6 +1760,11 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 			} else if !is_array_method_first_last_repeat && !(left_type.has_flag(.shared_f)
 				&& g.typ(left_type) == g.typ(node.receiver_type)) {
 				g.write('&')
+			}
+		} else {
+			if node.name != 'clone' && !left_type.is_ptr() {
+				g.write('ADDR(${rec_cc_type}, ')
+				cast_n++
 			}
 		}
 	} else if !node.receiver_type.is_ptr() && left_type.is_ptr() && node.name != 'str'

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1675,7 +1675,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 	// if so, then instead of calling array_clone(&array_slice(...))
 	// call array_clone_static(array_slice(...))
 	mut is_range_slice := false
-	if node.receiver_type.is_ptr() && !left_type.is_ptr() {
+	if node.name == 'clone' && node.receiver_type.is_ptr() && !left_type.is_ptr() {
 		if node.left is ast.IndexExpr {
 			idx := node.left.index
 			if idx is ast.RangeExpr {

--- a/vlib/v/tests/fns/method_call_on_rangeexpr_test.v
+++ b/vlib/v/tests/fns/method_call_on_rangeexpr_test.v
@@ -1,4 +1,4 @@
-type Buffer = []byte
+type Buffer = []u8
 
 pub fn (mut sb Buffer) vbytes() string {
 	return sb[0..sb.len].a().str()
@@ -9,7 +9,7 @@ pub fn (mut sb Buffer) a() Buffer {
 }
 
 fn test_main() {
-	mut b := Buffer([]byte{cap: 10})
+	mut b := Buffer([]u8{cap: 10})
 	b << 1
 	b << 2
 	assert b.vbytes() == 'Buffer([1, 2])'

--- a/vlib/v/tests/fns/method_call_on_rangeexpr_test.v
+++ b/vlib/v/tests/fns/method_call_on_rangeexpr_test.v
@@ -1,0 +1,16 @@
+type Buffer = []byte
+
+pub fn (mut sb Buffer) vbytes() string {
+	return sb[0..sb.len].a().str()
+}
+
+pub fn (mut sb Buffer) a() Buffer {
+	return sb
+}
+
+fn test_main() {
+	mut b := Buffer([]byte{cap: 10})
+	b << 1
+	b << 2
+	assert b.vbytes() == 'Buffer([1, 2])'
+}

